scheduler/scheduler-server/src/test/java/functionaltests/job/TagTest.java

```diff
@@ -0,0 +1,345 @@
+import static org.mockito.Mockito.mock;
+
+
+public class TagTest extends ProActiveTest{
+
+    /* mocks */
+    protected TaskResultImpl resultMock = mock(TaskResultImpl.class);
+    protected SchedulerStateUpdate schedulerStateUpdateMock = mock(SchedulerStateUpdate.class);
+    protected ExecuterInformation executerInformationMock = mock(ExecuterInformation.class);
+
+
+    protected InternalTaskFlowJob job = null;
+
+
+
+    @Before
+    public void setUp() {
+        job = new InternalTaskFlowJob();
+    }
+
+
+
+    private InternalScriptTask createTask(String name, InternalTask[] dependences, FlowBlock block, String matchingBlock)
+             {
+        InternalScriptTask result = new InternalScriptTask();
+        result.setName(name);
+
+        if(dependences != null && dependences.length > 0){
+            for(InternalTask dep: dependences){
+                result.addDependence(dep);
+            }
+        }
+
+        if(block != null){
+            result.setFlowBlock(block);
+            result.setMatchingBlock(matchingBlock);
+        }
+
+        job.addTask(result);
+        return result;
+    }
+
+
+
+    private InternalScriptTask createLoopTask(String name, String scriptContent, InternalTask[] dependences, String targetName, boolean block)
+            throws InvalidScriptException {
+        FlowBlock fb = null;
+        if(block){
+            fb = FlowBlock.END;
+        }
+        InternalScriptTask result = createTask(name, dependences, fb, targetName);
+        FlowScript loop = FlowScript.createLoopFlowScript(scriptContent, targetName);
+        result.setFlowScript(loop);
+
+        return result;
+    }
+
+
+    private InternalScriptTask createReplicateTask(String name, InternalTask[] dependences, FlowBlock block, String matchingBlock, int nbRuns)
+            throws InvalidScriptException {
+        InternalScriptTask result = createTask(name, dependences, block, matchingBlock);
+        FlowScript replicate = FlowScript.createReplicateFlowScript("runs = " + nbRuns + ";");
+        result.setFlowScript(replicate);
+        return result;
+    }
+
+
+
+    private void execute(InternalTask task){
+        task.setExecuterInformation(executerInformationMock);
+        job.startTask(task);
+        FlowScript script = task.getFlowScript();
+        FlowAction action = null;
+        if(script != null) {
+            action = task.getFlowScript().execute().getResult();
+        }
+        job.terminateTask(false, task.getId(), schedulerStateUpdateMock, action, resultMock);
+        System.out.println("executed " + task.getName() + " -> " + getTaskNameList(true));
+    }
+
+
+    private void execute(String [] tasks) throws UnknownTaskException {
+        for(String currentTaskName: tasks){
+            InternalTask currentTask = job.getTask(currentTaskName);
+            execute(currentTask);
+        }
+    }
+
+
+    private String getTaskNameList(boolean showTags){
+        StringBuffer buf = new StringBuffer();
```

Comment:
`getTaskNameList` seems not executed in parallel. If it is the case, _StringBuilder_ is enough.